src/progress.rs

```diff
@@ -175,16 +177,22 @@ pub struct ProgressSet {
     // You should not depend on these values unless you just set them.
     // We use a cell to avoid taking a `&mut self`.
     sort_buffer: RefCell<Vec<u64>>,
+    logger: Logger,
 }
 
 impl ProgressSet {
     /// Creates a new ProgressSet.
-    pub fn new() -> Self {
-        Self::with_capacity(0, 0)
+    pub fn new<'a>(logger: impl Into<Option<&'a Logger>>) -> Self {
```

Comment:
I wonder if having a builder `.with_logger()` would be nicer from an API point of view. 🤔